src/persist/src/indexed/encoding.rs

```diff
@@ -1537,4 +1538,40 @@ mod tests {
             Err(Error::from("duplicate batch key found in trace: "))
         );
     }
+
+    #[test]
+    fn encoded_batch_sizes() {
+        fn sizes(data: DataGenerator) -> (usize, usize) {
+            let unsealed = BlobUnsealedBatch {
+                desc: SeqNo(0)..SeqNo(1),
+                updates: data.batches().collect(),
+            };
+            let trace = BlobTraceBatch {
+                desc: Description::new(
+                    Antichain::from_elem(0),
+                    Antichain::from_elem(1),
+                    Antichain::from_elem(0),
+                ),
+                updates: data.records().collect(),
+            };
+            let (mut unsealed_buf, mut trace_buf) = (Vec::new(), Vec::new());
+            unsealed.encode(&mut unsealed_buf);
+            trace.encode(&mut trace_buf);
+            (unsealed_buf.len(), trace_buf.len())
+        }
+
+        let record_size_bytes = DataGenerator::default().record_size_bytes;
+        // Print all the sizes into one assert so we only have to update one
+        // place if sizes change.
+        assert_eq!(
+            format!(
+                "1={:?} 25={:?} 1000={:?} 1000/100={:?}",
```

Comment:
two small nits:

1. maybe it makes sense to also have one check in the millions of records here? I'm kind of undecided
2. the 1000/100 is meant to indicate "1000 records in 100 batches" right? I wonder if maybe that makes more sense with 1/1=... 25/1=... 1000/1... and then 1000/100. 

Feel free to take or leave both of these

Reply:
> maybe it makes sense to also have one check in the millions of records here? I'm kind of undecided

I gave it a shot but looks like the test takes way too long. Seems to be pretty close to 1000x the 1000 case, anyway.

> the 1000/100 is meant to indicate "1000 records in 100 batches" right? I wonder if maybe that makes more sense with 1/1=... 25/1=... 1000/1... and then 1000/100.

oops, it was supposed to be the raw params (records/records per batch) but I originally wrote the test to be 100, then later changed it to 10 and forgot to update this format string. your suggestion is better anyway, updated